src/golang/lib/livy/client_test.go

```diff
@@ -0,0 +1,173 @@
+package livy
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	mux    *http.ServeMux
+	server *httptest.Server
+	client *LivyClient
+)
+
+func setup() func() {
+	mux = http.NewServeMux()
+	server = httptest.NewServer(mux)
+
+	client = NewLivyClient(server.URL)
+
+	return func() {
+		server.Close()
+	}
+}
+
+func TestCreateSession(t *testing.T) {
+	cleanup := setup()
+	defer cleanup()
+
+	mux.HandleFunc("/sessions", func(w http.ResponseWriter, r *http.Request) {
+		jsonResp := `{"id": 1, "state": "idle"}`
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		w.Write([]byte(jsonResp))
+	})
+
+	// Call CreateSession with expected
+	sessionReq := &CreateSessionRequest{
+		Kind: "pyspark",
```

Comment:
does these test require an active spark cluster?

Reply:
No, these use the `httptest` package to mock the server.